api/src/main/java/org/pmiops/workbench/api/CohortAnnotationDefinitionController.java

```diff
@@ -105,13 +106,14 @@ public CohortAnnotationDefinition apply(org.pmiops.workbench.db.model.CohortAnno
         validateMatchingWorkspace(workspaceNamespace, workspaceId, cohort.getWorkspaceId());
 
         org.pmiops.workbench.db.model.CohortAnnotationDefinition cohortAnnotationDefinition =
-                cohortAnnotationDefinitionDao.findOne(annotationDefinitionId);
+                cohortAnnotationDefinitionDao.findByCohortIdAndAndCohortAnnotationDefinitionId(cohortId, annotationDefinitionId);
```

Comment:
What you had before is OK; you can just compare the cohortId on the definition that comes back to the cohort ID above... that might be simpler than adding the new method to the DAO like this.